feat: earn redesign

zhaono1 opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Earn redesign: Assets/Portfolio/FAQs tabs, protocols list, protocol details, Manage Position flows, shareable protocol links, TableList-driven lists, banner carousel, APY charts.

* **Enhancements**
  * Recommended tokens, network avatar display modes (overlapped/spread), improved breadcrumbs/navigation (modal→tab), WalletActionEarn logo support, MORPHO token/network support.

* **UI Improvements**
  * FAQ accordion, overview/rebate display, refreshed portfolio/asset layouts, per-row actions and claim flows, lightweight charts.

* **Bug Fixes**
  * Silenced noisy console logs in protocol list flow.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

zhaono1 committed:
feat: enhance Earn Home and Network Avatar components

zhaono1 committed:
feat: enhance Earn module with new components

zhaono1 committed:
feat: enhance Earn feature with new protocols page and navigation

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

revan-zhang commented:
### :white_check_mark: **Snyk checks have passed. No issues have been found so far.**

| Status | Scanner | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-critical.svg) Critical | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-high.svg) High | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-medium.svg) Medium | ![](https://res.cloudinary.com/snyk/image/upload/r-d/scm-platform/snyk-pull-requests/severity-low.svg) Low | Total (0) |
|---|---|---|---|---|---|---|
| :white_check_mark: | Open Source Security| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/b705d3c9-9b18-490d-8340-1563881fd7b3?refs=4f615521a10d2187b4e3ca09057658488ec64953&source=prChecksComment) |
| :white_check_mark: | Licenses| 0 | 0 | 0 | 0 | [ 0 issues](https://app.snyk.io/org/rayston92/pr-checks/b705d3c9-9b18-490d-8340-1563881fd7b3/license?refs=4f615521a10d2187b4e3ca09057658488ec64953&source=prChecksComment) |

:computer: Catch issues earlier using the plugins for [VS Code](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner), [JetBrains IDEs](https://plugins.jetbrains.com/plugin/10972-snyk-security), [Visual Studio](https://marketplace.visualstudio.com/items?itemName=snyk-security.snyk-vulnerability-scanner-vs-2022), and [Eclipse](https://marketplace.eclipse.org/content/snyk-security#external-install-button).

zhaono1 committed:
feat: refactor Earn module to improve navigation and UI componentsd logic.

zhaono1 committed:
feat: enhance Earn components with improved layout and logo URI handling

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
feat: enhance Earn and Staking features with new hooks and components

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
fix: rename loading prop to isAccountsLoading for clarity in Earn components

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
feat: add HeaderRight component for history action in ManagePosition page

zhaono1 committed:
feat: update TableList and Earn components for improved layout and responsiveness

zhaono1 committed:
feat: refactor Earn components and hooks for improved data handling and performance

zhaono1 committed:
feat: implement v2 investment and asset handling in Earn module

zhaono1 committed:
Merge remote-tracking branch 'origin/x' into feat/earn-redesign

zhaono1 committed:
feat: refactor Earn module to improve account and portfolio handling

zhaono1 committed:
feat: refactor Earn module components and hooks for improved structure and performance

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
feat: refactor ProtocolDetailsPage to improve data handling and remove unused actions

zhaono1 committed:
feat: add 'link' variant to Button component and refactor PortfolioTabContent for improved layout and currency formatting

zhaono1 committed:
feat: enhance staking and earn modules with airdrop support, metadata handling, and improved UI components

zhaono1 committed:
feat: add airdrop investment support and refactor related components for improved handling

zhaono1 committed:
feat: implement manage page functionality and enhance staking data handling

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
feat: add row press handling for portfolio items and optimize batch updates with RAF

zhaono1 committed:
feat: enhance staking functionality with new portfolio action handling and UI improvements

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

ByteZhang1024 committed:
chore: fix uniswap connect (#8903)

zhaono1 committed:
feat: enhance WrappedActionButton and usePortfolioAction to support reward token address for airdrop assets

huhuanming committed:
export EarnHomeWithProvider

sidmorizon committed:
fix: cloud backup bugs (#8904)

zhaono1 committed:
feat: add alerts handling to manage position page and update response type

ezailWang committed:
fix: some swap issue (#8905)

zhaono1 committed:
feat: add check for token price before estimating gas fee in claim function

franco-chan committed:
partial onboarding i18n & animation fine turning (#8906)

* added i18n

* i18n

* fixed import recovery phrase styles

zhaono1 committed:
feat: add approve information to protocol data in useManagePage hook

sidmorizon committed:
fix: transfer events (#8908)

zhaono1 committed:
feat: add IEarnSummaryV2 interface and getEarnSummaryV2 method to ServiceStaking
fix: handle optional token price in useUniversalClaim for gas fee estimation

huhuanming committed:
feat: integrate backup mnemonic page && add flow error monitoring (#8911)

ByteZhang1024 committed:
chore: fix image compression & clear code (#8912)

sidmorizon committed:
fix: optimize transfer events (#8910)

originalix committed:
feat: add IP table support for App client (#8907)

* feat: add SniConnect dependency and update Podfile sources

# Conflicts:
#	apps/mobile/ios/Podfile.lock

* feat: add IpRequest component to gallery and update routes

* chore: update dependencies and enhance IpRequest component

# Conflicts:
#	apps/mobile/ios/Podfile.lock

* chore: update SniConnect dependency to version 0.1.5 in Podfile.lock, package.json, and yarn.lock

* feat: implement IP Table adapter and SNI request handling for different platforms

* feat: add IgnorePlugin for react-native-sni-connect and refactor SNI support check in IP Table adapter

* fix: filter out undefined and null values from axios request parameters in IP Table adapter

* feat: implement debug logging in IP Table adapter for better development insights

* feat: implement IP Table service and configuration management with lazy loading in SimpleDb

* feat: add ip-table dev settings

* fix: improve IP Table usage checks in development and production environments

* feat: implement SNI request handling in DesktopApi with Electron net module

* feat: refactor SNI request handling to use Node.js https module instead of Electron net

* refactor: remove unnecessary logging from SNI request handling

* feat: implement custom HTTPS agent for connection reuse in SNI requests

* fix: disable debug logging in production environment

* feat: add response decompression support for SNI requests

* feat: add additional skip words for spell checker

* chore: update Podfile.lock

* feat: enhance IP Table configuration with CDN support and runtime state management

* feat: implement IP Table speed test functionality and SNI failure reporting

* feat: update IP Table default configuration with new endpoints and metadata

* feat: implement remote IP Table configuration fetching and merging functionality

* feat: add unit tests for IP Table configuration signature verification and merging functionality

* feat: enhance IP Table functionality with platform support checks and update signer address verification

* feat: lazy load simpleDb in updateInterceptorRequestHelper to optimize module initialization

* chore: use @onekey/react-native-sni-connect

* feat: refactor SNI request handling and improve platform compatibility

* chore: lint

* feat: implement structured logging for IP Table operations and add IpTableScope

* feat: add forceIpTableStrict setting and implement strict mode fallback for IP selection

* feat: memoize IP selection logic for performance optimization

* feat: enhance SNI failure handling with cooldown and failure counting

ByteZhang1024 committed:
Feat/bifrost dot (#8914)

* feat: support bifost dot

* feat: remove dot

* fix: lint

weatherstar committed:
optimize: non backed up empty view & receive token modal OK-44402 OK-45540 (#8916)

* feat: referral code block

* feat: receive info block

* fix: help link in receive selector

* feat: add cloud back up entry in home

* fix: update help link

* fix: add money layout

* fix: lint

* fix: ui

* ui & i18n

* fix: receive selector modal

* fix: params

* refactor: move add money block to top

* fix: add money block position

* fix: receive selector router

* ui

* ui

* fix: lint

---------

Co-authored-by: Franco <franco@onekey.so>

ezailWang committed:
fix: add perp deposit event (#8913)

Signed-off-by: ezailWang <jelly@onekey.so>
Co-authored-by: morizon <sidmorizon@outlook.com>

zhaono1 committed:
feat: enhance Overview component with rebate data handling and display

zhaono1 committed:
fix: improve account existence check in useEarnAccounts hook

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
fix: enhance handleHistoryPress to fetch current earn account details

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
fix: improve layout and responsiveness in PortfolioTabContent and TableList components

zhaono1 committed:
feat: add event bus support for refreshing earn portfolio items

zhaono1 committed:
feat: add portfolioSymbol to claim actions and refresh logic

zhaono1 committed:
feat: refactor EarnMainTabs and PortfolioTabContent for improved mobile layout and responsiveness

zhaono1 committed:
feat: add event bus support for refreshing earn portfolio and improve refresh logic

zhaono1 committed:
feat: add APY history response interfaces and implement getApyHistory method in ServiceStaking

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
feat: Implement ApyChart component for displaying APY history

zhaono1 committed:
feat: EarnProtocolDetails share

zhaono1 committed:
feat: refactor EarnProtocolDetails to improve navigation and add footer support

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
feat: enhance portfolio actions with stakedSymbol support for airdrop claims

zhaono1 committed:
feat: add validation for required data in StakeSection and WithdrawSection components

zhaono1 committed:
feat: i18n

zhaono1 committed:
feat: improve loading state handling in PortfolioTabContent

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
feat: adjust padding for Overview component based on screen size

zhaono1 committed:
feat: add account and address validation to disable action buttons in EarnProtocolDetailsPage

zhaono1 committed:
feat: update navigation for Earn protocols and enhance conditional rendering in components

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

originalix commented on packages/kit/src/views/Earn/components/BannerItemV2.tsx:
```diff
@@ -0,0 +1,55 @@
+import { Image, SizableText, Stack, useMedia } from '@onekeyhq/components';
+import type { IDiscoveryBanner } from '@onekeyhq/shared/types/discovery';
+
+const BANNER_TITLE_OFFSET = {
```
这类没用到的变量都可以清理下，还有下面的 media

originalix commented on packages/kit/src/views/Earn/components/PortfolioTabContent.tsx:
```diff
@@ -0,0 +1,704 @@
+
+  return (
+    <YStack>
+      {investments.map((item, index) => {
```
这个写法，会不会有长列表的滚动问题？

originalix commented on packages/kit/src/views/Earn/components/Recommended.tsx:
```diff
@@ -0,0 +1,326 @@
+  token,
+  ...rest
+}: { token?: IRecommendAsset } & IYStackProps) {
+  const accountInfo = useActiveAccount({ num: 0 });
```
事件和账户信息都从父组件传入，减少子组件的事件绑定数量，和 hooks 变动导致的重渲染

originalix commented on packages/kit/src/views/Earn/hooks/useBannerInfo.ts:
```diff
@@ -0,0 +1,29 @@
+export const useBannerInfo = () => {
+  const { result: earnBanners, run: refetchBanners } = usePromiseResult(
+    async () => {
+      const bannerResult =
```
请求的接口是 /earn-banner/list，但是函数名的 EarnHomeData，不够语义化

originalix replied:
banner 的数据最好也放在 simpleDb 里，通过 atom 在初始化时写入，首次渲染时先加载缓存数据

originalix commented on packages/kit/src/views/Earn/hooks/useAllNetworkId.ts:
```diff
@@ -0,0 +1,6 @@
+
+import { getNetworkIdsMap } from '@onekeyhq/shared/src/config/networkIds';
+
+export const useAllNetworkId = () =>
```
感觉没必要增加一个 hook，代码里直接 `const allnetworkId = getNetworkIdsMap().onekeyall;` 也很直观

originalix commented on packages/kit/src/views/Staking/pages/ManagePosition/hooks/useEarnAccount.ts:
```diff
@@ -0,0 +1,25 @@
+import backgroundApiProxy from '@onekeyhq/kit/src/background/instance/backgroundApiProxy';
+import { usePromiseResult } from '@onekeyhq/kit/src/hooks/usePromiseResult';
+
+export const useEarnAccount = ({
```
两个同名的 hooks 了

originalix reviewed (commented):
移动端进入 2 级页面，得像 Market 一样隐藏下 TabBar？跟 pm 确认下

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 commented:
> 移动端进入 2 级页面，得像 Market 一样隐藏下 TabBar？跟 pm 确认下

ok tabbar 这期间应该改版过，我问下我这边咋适配

zhaono1 committed:
feat: add recommended tokens functionality to earn context and components

zhaono1 committed:
feat: add banner management functionality to earn context and components

zhaono1 committed:
feat: rename fetchEarnHomePageData to fetchEarnHomePageBannerList for clarity

zhaono1 committed:
feat: replace useAllNetworkId hook with direct getNetworkIdsMap usage for network ID retrieval

zhaono1 committed:
feat: refactor RecommendedItem to use memoization and pass account details as props

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
feat: update investment filtering to exclude zero-value investments, including airdrops

zhaono1 committed:
feat: refactor ApyChart to track mouse position for popover and add PerformanceSection to display performance details

zhaono1 committed:
feat: enhance Earn and Staking components with new header and footer structures, improve layout responsiveness

zhaono1 committed:
feat: implement LightweightChart component

zhaono1 committed:
feat: enhance Earn and ManagePosition components to support tab navigation and improve routing parameters

zhaono1 committed:
feat: earn i18n

zhaono1 committed:
feat: add useProtocolDetailBreadcrumb

zhaono1 committed:
feat: enhance Earn and ManagePosition components with improved loading states and withdraw action handling

zhaono1 committed:
feat: wrap Overview component in XStack for improved layout

zhaono1 committed:
feat: update route types and add new action icon for Earn details

zhaono1 committed:
feat: remove revalidateOnFocus option and update accountId reference in ManagePosition

zhaono1 committed:
feat: add popover position calculation and update accountId reference in ManagePosition

zhaono1 committed:
feat: refactor ManagePosition components for improved structure and readability

zhaono1 committed:
feat: implement USDe handling in ManagePosition components and related hooks

zhaono1 committed:
feat: refactor ManagePosition and ManagePositionContent for USDe protocol support and improved structure

zhaono1 committed:
feat: remove unused accordion trigger text in UniversalStake and UniversalWithdraw components

zhaono1 committed:
feat: update layout in Overview component for improved responsiveness

zhaono1 committed:
feat: add rewardSymbol support in portfolio actions and hooks for improved airdrop handling

zhaono1 committed:
feat: enhance navigation handling in ManagePositionContent for modal contexts

zhaono1 committed:
feat: enhance withdrawal navigation handling in ManagePositionContent for modal contexts

zhaono1 committed:
feat: refactor navigation handling in ManagePositionContent, StakeSection, and WithdrawSection for improved modal context support

zhaono1 committed:
feat: enhance ManagePosition and Withdraw sections with beforeFooter prop and improve alert handling

zhaono1 committed:
feat: redesign earnings management interface and enhance portfolio handling with new components and improved alert system

zhaono1 committed:
feat: adjust layout of DepositField to improve responsiveness with maxWidth and flexWrap properties

zhaono1 committed:
feat: improve deposit action text handling for better type safety

zhaono1 committed:
feat: remove maxWidth constraints from DepositField and EarningsField for improved layout flexibility

zhaono1 committed:
Merge remote-tracking branch 'origin/x' into feat/earn-redesign

zhaono1 committed:
feat: add alerts handling to ManagePositionContent and useManagePage for improved user notifications

zhaono1 committed:
feat: enhance layout and styling in EarnPageContainer and PortfolioTabContent for improved readability

zhaono1 committed:
feat: add custom header right items and share functionality to EarnPageContainer and ApyChart components

zhaono1 committed:
feat: add alertsHolding and activate action to ManagePositionContent and USDEManageContent for enhanced user interaction

zhaono1 committed:
feat: add showApyDetail prop to various components for enhanced APY detail visibility

zhaono1 committed:
feat: add ProtectionSection to DetailsPart for enhanced detail visibility

zhaono1 committed:
feat: refactor account handling and remove unused protocol details hook for improved code clarity

zhaono1 committed:
feat: add isWithdrawOrder logic to handle tab changes based on withdrawal state

zhaono1 committed:
feat: update provisioning profile for OneKeyWallet ServiceExtension to reflect new development date

zhaono1 committed:
feat: update provisioning profile date and clean up linker flags for improved build configuration

zhaono1 committed:
feat: update linker flags format for improved build configuration

zhaono1 committed:
feat: add withdraw order handling to navigate directly to WithdrawOptions modal

zhaono1 committed:
feat: add autoToast option to BTCFreshAddressCanNotConnectDappError for better user feedback

zhaono1 committed:
feat: integrate toastIfError decorator for improved error handling in ServiceStaking and update BTCFreshAddressCanNotConnectDappError usage

zhaono1 committed:
feat: enhance investment handling and action buttons in Earn module

zhaono1 committed:
feat: refactor Overview and PortfolioTabContent components for improved code clarity and structure

zhaono1 committed:
feat: refactor Earn module to use useEarnPortfolio hook and improve portfolio data handling

zhaono1 committed:
feat: enhance refresh logic in BasicEarnHome to include portfolio data refresh

zhaono1 committed:
feat: optimize investment data fetching by processing results in bulk

zhaono1 committed:
feat: add isInModalContext prop to various components for modal context handling

zhaono1 committed:
feat: add AlertSection to DetailsPart for improved alert display

zhaono1 committed:
feat: add isInModalContext prop to ManagePositionContent and USDEManageContent for modal handling

zhaono1 committed:
feat: update ActionField layout for better responsiveness and structure

zhaono1 committed:
feat: add MORPHO support in staking configuration and address constants

zhaono1 committed:
feat: enhance useProtocolDetailData to conditionally include account info

zhaono1 committed:
feat: add KYC account address handling for Ethena provider and refactor useProtocolDetailData

zhaono1 committed:
feat: add token image URI handling across ManagePosition components and related pages

zhaono1 committed:
feat: refactor account ID handling in useEarnPortfolio hook for improved clarity

zhaono1 committed:
feat: enhance useEarnPortfolio to support streaming updates and deduplication during data fetch

zhaono1 committed:
feat: simplify banner rendering and enhance layout in Earn components

zhaono1 committed:
feat: refactor PortfolioTabContent and BasicEarnProtocols for improved layout and readability

zhaono1 committed:
feat: add headerProps support to TableList and TableListHeader for enhanced customization

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
feat: update ProtocolHeader styles for better responsiveness and adjust Divider margins

huhuanming committed:
fix: correctly identify which hardware needs updating when multiple devices are connected (#9040)

sidmorizon committed:
Fix/revenuecat privy logs (#9038)

* fix: logout log

* fix: invlid token log

* fix: websocket logout event log

* fix: atom not login log

* fix: privy log

* fix: revenuecat log

yikZero committed:
fix: update SocialButtonGroup styling for better alignment and text color consistency (#9036)

zhaono1 committed:
feat: enhance loading skeletons in PortfolioTabContent and EarnProtocols for improved UI consistency

zhaono1 committed:
feat: add logoURI prop to WalletActionEarn and TokenDetailsHeader for enhanced token representation

originalix committed:
feat: add support for desktop USB transport type in onboarding process (#9041)

zhaono1 committed:
feat: improve loading state handling in PortfolioTabContent and remove unnecessary withdrawDisabled check in NormalManageContent

zhaono1 committed:
feat: add isInModalContext prop to UniversalWithdraw for modal compatibility

zhaono1 committed:
feat: initialize investment state with total values from current overview data

sidmorizon committed:
fix: prime device limit alert (#9043)

zhaono1 committed:
feat: simplify loading and empty state logic in PortfolioTabContent

limichange committed:
refactor: enhance referral components with responsive design and new phase actions (#9042)

- Added responsive behavior to the ReferFriendsPageContainer using `useMedia`.
- Introduced ReferAFriendPhaseActions component for managing phase transitions.
- Updated ReferAFriendPage to conditionally render actions based on screen size.
- Removed unused JoinButton and NextButton components to streamline code.
- Adjusted layout in various components for improved user experience and consistency.

ezailWang committed:
fix: swap toaccount issue fix (#9044)

zhaono1 committed:
feat: remove loading skeletons from ApyChart, ManagePositionContent, and StakeSection for improved performance

originalix committed:
feat: implement IP Table enablement checks in ServiceIpTable (#9045)

* feat: implement IP Table enablement checks in ServiceIpTable

* fix: prevent redundant network diagnostics and handle cleanup on result page

franco-chan committed:
improve onboarding performance & unlock page loading state (#9048)

ezailWang committed:
fix: swap to address catch error (#9047)

zhaono1 committed:
feat: refactor WrappedActionButton and add ProtocolAirdrop component for improved airdrop handling

zhaono1 committed:
feat: update placement of action icon tooltip to bottom-start for better visibility

zhaono1 committed:
feat: enhance rebate data check to handle empty distributed and undistributed cases

huhuanming committed:
fix: restore OneKey ID header navigation bar(OK-46269) (#9055)

zhaono1 committed:
feat: update token structure in portfolio actions to include price and info

zhaono1 committed:
feat: add earnPortfolioInvestmentsAtom and enhance useEarnPortfolio hook for investment management

ByteZhang1024 committed:
fix: google play submit error (#9056)

originalix committed:
fix: enhance BLE onboarding by building USB connectId from device features (#9058)

zhaono1 committed:
feat: add unique keys to CalculationListItem and other components for improved rendering

ezailWang committed:
fix: perp deposit mixpannel event token object opt (#9051)

zhaono1 committed:
feat: optimize investment rendering and add event listener for account creation

weatherstar committed:
fix: receive selector style & support token param (#9053)

* chore: update locale

* fix: receive info block style

* feat: pass token to receive selector

* fix: lint

* fix: receive style

zhaono1 committed:
feat: update event listener for account data changes to refresh portfolio data

zhaono1 committed:
feat: update EarnTooltip and related components to handle tooltip descriptions

zhaono1 committed:
fix: add optional chaining to tooltip data access in UniversalStake and UniversalWithdraw components

zhaono1 committed:
refactor: simplify tab content rendering in EarnMainTabs component

zhaono1 committed:
fix: update state setting in ContextJotaiActionsEarn to use a function for payload
delete: remove unused ApproveBaseStake component and its related code

zhaono1 committed:
feat: add loading skeleton to ManagePositionContent and improve data fetching logic

zhaono1 committed:
Merge branch 'x' of github.com:OneKeyHQ/app-monorepo into feat/earn-redesign

zhaono1 committed:
fix: optimize BannerV2 and FAQContent components for better media responsiveness

zhaono1 committed:
fix: simplify SectionSkeleton layout by removing unnecessary padding and skeleton elements

zhaono1 committed:
fix: refactor account state management to streamline request handling

zhaono1 committed:
fix: enhance BannerV2 component with optimized renderItem logic and improve PortfolioTabContent key extraction

zhaono1 committed:
fix: enhance portfolio data handling with event registration and memoization

zhaono1 committed:
fix: enhance memory management and error handling across components

zhaono1 committed:
fix: improve account fetching logic and loading states in Earn components

zhaono1 committed:
fix: enhance performance and memory management in Earn components with memoization and cleanup logic

zhaono1 committed:
fix: clean up unused imports and improve route handling in Earn components

zhaono1 committed:
refactor: streamline navigation and remove unused token provider logic in Earn components

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
fix: update navigation logic in safePushToEarnRoute function

zhaono1 committed:
fix: adjust rotation direction for expanded state in TableListRow and set width for YStack in ProtocolAirdrop

zhaono1 committed:
feat: add Earn pages to Discovery tab and enhance navigation logic in safePushToEarnRoute

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

zhaono1 committed:
refactor: improve route handling and streamline navigation in Earn components

zhaono1 committed:
Merge branch 'x' into feat/earn-redesign

originalix committed:
Merge branch 'x' into feat/earn-redesign

originalix commented:
@coderabbitai resolve
